scripts/linux/publishReleasePackages.sh

```diff
@@ -238,8 +238,10 @@ publish_to_github()
         sed -i "$ d" $WDIR/content.txt
 
         #Create Release Page
+        #  This unintentionally create a github tag associate with the latest commit which is an undesired behavior. 
+        #  We will need to update the tag to the proper commit once the /azure-iotege latest json are ready to go.
```

Comment:
Are you saying that this will create a tag in the upstream repo pointing to the wrong commit, and that we'll need to fix the tag later in the release process to point to the right commit?

It's ok to "move" tags in your local repo, but once a tag is in GitHub it shouldn't change. See the [git documentation](https://git-scm.com/docs/git-tag?msclkid=e1ff496ecd8211ecb0ce20073c42cd8d#_on_re_tagging) for a discussion about re-tagging.

Reply:
> Are you saying that this will create a tag in the upstream repo pointing to the wrong commit, and that we'll need to fix the tag later in the release process to point to the right commit?

That is correct. This API directly call the upstream repository and TAG it in the process. The github API requires a tag in order to create a release page. I cannot just create a release page and promise that a tag will be available. 

We are running into a catch-22 here where as soon as you commit the merge latest version json in /azure-iotedge repository, you're essentially update all the pointer in iotedge/docker that the latest version is out; but the latest version is really not out fully out until the github release page is published.

How do you feel if I do a temporary tag say... `<version>-draft` . The tag will be used during the release process to create a draft release page. Then once the whole release process is completed, we create a new tag `<version>` and remove the `<version>-draft` tag from the repo?

Reply:
I just used `POST /repos/{owner}/{repo}/releases` to create a Release in a private repo, with `draft: true`. I specified a tag, but when the draft Release was created _it did not create the tag_. Then I published the release with `PATCH /repos/{owner}/{repo}/releases/{release_id}` (payload: `{"target_commitish":"xxxxxxx","draft":false}`) and the tag was created, pointing to commit xxxxxxx.

I think you can remove the first line of the comment above ("# This unintentionally...") and keep the second line as a TODO.

Reply:
Thanks for the investigation. Let me give it a shot.

Reply:
I'm a bit confused here: 

What you're saying makes complete sense Damon. But if the page is created as a draft, then there's no harm done because it's not actually published, right? So no customer has actually pulled the release..?

And when you say that no tag was created with `draft: true` in the private repo, what was the difference in your testing vs Bear's?